pymc3/sampling.py

```diff
@@ -1208,6 +1209,49 @@ def sample_ppc_w(traces, samples=None, models=None, weights=None,
     return {k: np.asarray(v) for k, v in ppc.items()}
 
 
+def sample_generative(samples=500, model=None, vars=None, random_seed=None):
```

Comment:
should we rename this `sample_prior_predictive`?

Reply:
and `sample_ppc` should then probably be `sample_posterior_predictive` (it's not really doing a check).

Reply:
I think this is now actually the generative function (see the test case).  The prior predictive would be nice to also implement.

Reply:
I looked at the test but I don't understand the difference between generative and prior predictive here.

Reply:
Ugh... I think I still don't.  Will work out a simple model by hand later to try to help my intuition.